### PR TITLE
Update ApplePayJs type to include Recurring payment support.

### PR DESCRIPTION
**Pull request by clates**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

----

Ran prettier as well as updated the tests to include the examples given by the [ApplePayDemo site](https://applepaydemo.apple.com).  Comments, type descriptions, and types were taken from  [the Apple documentation on Apple Pay JS API](https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api)

### CODE REVIEW

File: types/applepayjs/index.d.ts [hunk 1]
```diff
@@ -658,6 +769,74 @@ declare namespace ApplePayJS {
...
+     * Use an ApplePayRecurringPaymentRequest object to provide the user with payment details and a way to manage payment methods for a recurring payment.
+     * You can optionally display a billing agreement and set up merchant token life-cycle notifications for the request.
+     */
+    interface ApplePayRecurringPaymentRequest {
+        paymentDescription: string;
+        regularBilling: ApplePayLineItem;
+        trialBilling?: ApplePayLineItem;
+        billingAgreement?: string;
+        managementURL: string;
+        tokenNotificationURL?: string;
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:809]: Could you add the docs for the individual members too?

File: types/applepayjs/index.d.ts [hunk 2]
```diff
@@ -658,6 +769,74 @@ declare namespace ApplePayJS {
...
+     *
+     * Apple Pay issues an Apple Pay Merchant Token if the user’s payment network supports merchant-specific payment tokens.
+     * Otherwise, Apple Pay issues a device token for the payment request.
+     */
+    interface ApplePayAutomaticReloadPaymentRequest {
+        paymentDescription: string;
+        automaticReloadBilling: ApplePayLineItem;
+        billingAgreement?: string;
+        managementURL: string;
+        tokenNotificationURL?: string;
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:825]: +1

File: types/applepayjs/index.d.ts [hunk 3]
```diff
@@ -658,6 +769,74 @@ declare namespace ApplePayJS {
...
+     * To enable multiple merchants for a transaction, use one ApplePayPaymentTokenContext object for each merchant.
+     *
+     * You can optionally associate each payment token with the merchant’s top-level domain.
+     */
+    interface ApplePayPaymentTokenContext {
+        merchantIdentifier: string;
+        externalIdentifier: string;
+        merchantName: string;
+        merchantDomain?: string;
+        amount: string;
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:839]: +1

File: types/applepayjs/index.d.ts [hunk 4]
```diff
@@ -738,6 +917,32 @@ declare namespace ApplePayJS {
...
+         */
+        dateComponentsRange?: ApplePayDateComponentsRange;
+    }
+
+    /**
+     * A dictionary that specifies the start and end dates for a range of time.
+     */
+    interface ApplePayDateComponentsRange {
+        startDateComponents: ApplePayDateComponents;
+        endDateComponents: ApplePayDateComponents;
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:932]: +1

File: types/applepayjs/index.d.ts [hunk 5]
```diff
@@ -738,6 +917,32 @@ declare namespace ApplePayJS {
...
+     * When specifying a range using date components, provide all elements of the ApplePayDateComponents down to the level of granularity that you want to expose.
+     * For example, if you specify a range of days, be sure to include values for both months and years in addition to days in the ApplePayDateComponents.
+     *
+     * Apple Pay on the Web uses the Gregorian calendar when processing dates.
+     */
+    interface ApplePayDateComponents {
+        days: number;
+        months: number;
+        years: number;
+        hours: number;
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:945]: +1

**Review by martincostello — COMMENTED** [review]
LGTM other than adding some additional `/**/` docs.

**Comment by clates** [comment]
@martincostello - Thanks for the 👀  - Addressed in 373eb88

File: types/applepayjs/index.d.ts [hunk 6]
```diff
@@ -658,6 +769,135 @@ declare namespace ApplePayJS {
...
+        paymentDescription: string;
+
+        /**
+         * A localized billing agreement that the payment sheet displays to the user before the user authorizes the payment.
+         */
+        billingAgreement?: string;
+
+        /**
+         * A localized billing agreement that the payment sheet displays to the user before the user authorizes the payment.
+         */
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:816]: This comment is copy-paste of the one above.

**Review by martincostello — APPROVED** [review]

**Pull request merged**